internal/internal.go

```diff
@@ -164,6 +164,10 @@ var (
 
 	// ORCAAllowAnyMinReportingInterval is for examples/orca use ONLY.
 	ORCAAllowAnyMinReportingInterval interface{} // func(so *orca.ServiceOptions)
+
+	// GRPCResolverSchemeExtraMetadata determines when gRPC will add extra
+	// metadata to RPCs.
+	GRPCResolverSchemeExtraMetadata string = "xds"
```

Comment:
Are we supporting `xdstp` for the target string?  If so, this needs to be extended.

This is read here when doing metadata matching for routes: https://github.com/grpc/grpc-go/blob/2aa261560586eab6795301a3670d9dfdd7308625/xds/internal/xdsclient/xdsresource/matcher.go#L100

Reply:
> Are we supporting `xdstp` for the target string?

Hmm ... [A47 section on target-uri syntax](https://github.com/grpc/proposal/blob/master/A47-xds-federation.md#grpc-client-target-uri-syntax) says we will continue to support the `xds` scheme for convenience. It does not say that we will start supporting the `xdstp` scheme. Nor does it say we will not support the `xdstp` scheme and instead the translations will be handled by the bootstrap configuration.

Do you think we should improve the language in this gRFC section or is it clear to you?

Reply:
It's not super air-tight, but by not saying "we're supporting `xdstp` as a gRPC name resolver scheme and here's how", then I wouldn't assume we are adding it.  We'd need a spec for how to handle it as a channel-level target.  The text that follows does describe how we'll support this new stuff in the existing `xds` resolver instead of adding a top-level resolver scheme for `xdstp`:

> we want the deployment to be able to decide to use new-style resource names without requiring users to explicitly specify them.  To that end, we will add support for an optional authority in the `xds` URI.  The procedure for handling an `xds` URI is as follows: